examples/hermes-engine/HermesRuntime.cs

```diff
@@ -100,45 +105,65 @@ public void CreatePolyfills()
 
         global["setImmediate"] = (JSCallback)(args =>
         {
-            JSValue immediateCallback = args[0];
-            VerifyElseThrow(immediateCallback.TypeOf() == JSValueType.Function,
-                "Wrong type of args[0]. Expects a function.");
-
+            JSValue immediateCallback = AsFunction(args, 0);
             int taskId = AddImmediateTask(immediateCallback);
             return taskId;
         });
 
         global["clearImmediate"] = (JSCallback)(args =>
         {
-            RemoveImmediateTask((int)args[0]);
+            int taskId = AsInt32(args, 0);
+            RemoveImmediateTask(taskId);
             return default;
         });
 
         global["setTimeout"] = (JSCallback)(args =>
         {
-            JSValue timeoutCallback = args[0];
-            VerifyElseThrow(timeoutCallback.TypeOf() == JSValueType.Function,
-                "Wrong type of args[0]. Expects a function.");
-
-            int taskId = AddTimerTask(timeoutCallback, (int)args[1]);
+            JSValue timeoutCallback = AsFunction(args, 0);
+            int delayInMs = AsInt32(args, 1);
+            int taskId = AddTimerTask(timeoutCallback, delayInMs);
             return taskId;
         });
 
         global["clearTimeout"] = (JSCallback)(args =>
         {
-            RemoveTimerTask((int)args[0]);
+            int taskId = AsInt32(args, 0);
+            RemoveTimerTask(taskId);
             return default;
         });
 
         var console = new JSObject();
         console["log"] = (JSCallback)(args =>
         {
-            Console.WriteLine((string)args[0]);
+            Console.WriteLine((string)args[0].CoerceToString());
```

Comment:
`JSValue.ToString()` already calls the JS `toString()`, so I think the cast / coercion is not necessary.